src/components/Timer/Timer.jsx

```diff
@@ -62,6 +71,9 @@ function Timer() {
     () => {
       if (seconds) {
         setSeconds(seconds - 1);
+        if (isSoundOn && seconds === 3) {
+          countdownSound.play();
+        }
```

Comment:
`seconds` uzun süre `3` 'e denk olmayacak bu yüzden bu olana kadar bu koşula sürekli `isSoundOn` u kontrol ettirmek gereksiz olabilir. Örneğin `1000` saniyen var `997` key `isSoundOn` un değeri `true` mu diye bakacak. `true` ise bile `seconds` un değeri `3` olmadığı sürece çalışmayacak.

```js
if (seconds === 3 && isSoundOn) {
       countdownSound.play();
}
```

Ufak bir sıra değişikliği ile daha iyi çalışır. `&&` olduğu için `second` 3 olmadığı sürece `isSoundOn` u kontrol etmez.